src/cluster/Telemetry.cc

```diff
@@ -131,28 +127,41 @@ void VerboseTelemetry::OnIncomingEvent(std::string_view topic, std::string_view
 
 namespace {
 
-std::string determine_script_location() {
-    std::string result = "none";
-
-    if ( zeek::detail::call_stack.empty() )
-        return result;
+// Cache CallExpr pointers to their location.
+std::map<const zeek::detail::CallExpr*, std::string> location_cache;
 
+// Cached lookup of a script location.
+std::string_view determine_script_location() {
     ssize_t sidx = static_cast<ssize_t>(zeek::detail::call_stack.size()) - 1;
-
     while ( sidx >= 0 ) {
         const auto* func = zeek::detail::call_stack[sidx].func;
         const auto* ce = zeek::detail::call_stack[sidx].call;
 
-        // without_zeekpath_component looks pretty expensive and might be
-        // better to cache the result using the ce pointer instead of computing
-        // it over and over again.
+        // Cached?
+        auto it = location_cache.find(ce);
+        if ( it != location_cache.end() )
+            return it->second;
```

Comment:
I'd try to avoid reusing `it` later and instead write

```suggestion
        if ( auto it = location_cache.find(ce); it != location_cache.end() )
            return it->second;
```